tokenizers.h

```diff
@@ -3,7 +3,10 @@
 
 struct Buffer {
   uint32_t *ids;
+  uint32_t *type_ids;
   char *tokens;
+  uint32_t *special_tokens_mask;
+  uint32_t *attention_mask;
```

Comment:
I kind of regret adding this struct to expose `char *tokens`. It made encode call about ~2x slower due to extra allocations and most users don't care about string tokens. For your feature, can you make it a distinct rust/c method for encode? The go interface could still be the same (perhaps with additional flags) to request one implementation over the other, however that way users that don't use these features won't be slowed down by this extra allocations.

Reply:
I agree, when writing this code I wondered whether we should have some way of parameterizing what gets back to the Go API so that you only pay the price for exactly what you need. I propose to introduce a new EncodeWithOptions function to which we can pass the list of attributes (as options) in order parameterize which attributes are to get retrieved. Would that be ok?

Reply:
Obviously we would keep the Encode function as it was before this change to not impact its performance.

Reply:
Hello, I think this makes a lot of sense. For instance, in our case we needed not only the attention mask, but also the token_type_ids to be able to use an onnx version of the MiniLM model. So we essentially forked the repo and just added rust code to the encode method to return those and the attention mask. It would be great to have an EncodeWIthOptions to customize the output

Reply:
A similar thing would be handy when reading the tokenizer in, e.g., to set truncation and padding. Maybe the from_file and from_bytes methods could support an option struct?

Reply:
I'll update the PR with the EncodeWithOptions very soon.

Regarding the configuration of the tokenizer, may I suggest you open a new dedicated issue to discuss that topic specifically?

Reply:
@daulet , @riccardopinosio , done. Let me know what you think.

Reply:
@clems4ever looks great to me.
@daulet can we merge this one?

Reply:
Sorry this was buried in my github notifications.

I like how you were able to reuse the same method while avoiding allocations. Do you mind resolving conflicts and rebasing? I added CI so we can quickly run the tests.

Reply:
@clems4ever just making sure you saw this, would love to have this in the main repo, then I can kill off my fork

Reply:
PR rebased and benchmark results posted in the PR